modules/sequenced_capture_executor.py

```diff
@@ -691,9 +690,19 @@ def _capture(
                             image = image_utils.add_false_color(array=image, color=use_color)
 
                         image = np.flip(image, 0)
-                        video_writer.add_frame(image=image)
+                        video_images.append((image, datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")[:-3]))
```

Comment:
The issue that we were having before-hand is that we didn't know exactly how many frames would be captured in the capture duration. We were expecting the fps, but it was different, leading to the video being shortened or taking longer to capture. By storing all of the images first, then passing them through the video_writer, we know exactly how many frames were taken during the duration, so we can create a video with the known fps and ensure that it ends up being the correct length.

Reply:
Right now, each image is being stored in a tuple (with its timestamp) inside of a list. Do you see any possible issues that might arise with this method? It can be a max of 900 images in the worst-case scenario but is usually much less than that. Would there be any other data structure or method that might be a better route, or does a list work fine?

Reply:
A list is good.  900 images might eat up a good chunk of RAM, but OpenCV might have been doing that buffering anyway in RAM, so may not make a difference.

Reply:
I wouldn't string-format the timestamp at this point, I would let the video_writer do that (like it was).  Should be able to store the datetime object in the tuple just like a string.